openshift/build-tool-detector.app.yaml

```diff
@@ -0,0 +1,121 @@
+  metadata:
+    labels:
+      service: build-tool-detector
+    name: build-tool-detector
+  spec:
+    replicas: ${{REPLICAS}}
+    selector:
+      service: build-tool-detector
+    strategy:
+      rollingParams:
+        intervalSeconds: 1
+        maxSurge: 25%
+        maxUnavailable: 25%
+        timeoutSeconds: 600
+        updatePeriodSeconds: 1
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          service: build-tool-detector
+      spec:
+        containers:
+        - image: ${IMAGE}:${IMAGE_TAG}
+          env:
+          - name: BUILD_TOOL_DETECTOR_GITHUB_CLIENT_ID
+            valueFrom:
+              secretKeyRef:
+                name: build-tool-detector
+                key: github.client.id
+          - name: BUILD_TOOL_DETECTOR_GITHUB_CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: build-tool-detector
+                key: github.client.secret
+          - name: BUILD_TOOL_DETECTOR_SENTRY_DSN
+            valueFrom:
+              secretKeyRef:
+                name: build-tool-detector
+                key: github.sentry.dsn
+          - name: BUILD_TOOL_DETECTOR_AUTH_URI
+            valueFrom:
+              configMapKeyRef:
+                name: build-tool-detector
+                key: auth.uri
+          imagePullPolicy: Always
+          name: build-tool-detector
+          ports:
+          - containerPort: 8099
+            protocol: TCP
+          livenessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /api/status
+              port: 8099
+              scheme: HTTP
+            initialDelaySeconds: 1
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /api/status
+              port: 8099
+              scheme: HTTP
+            initialDelaySeconds: 1
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          resources:
+            requests:
+              cpu: 0.1m
+              memory: 10Mi
+            limits:
+              cpu: 400m
+              memory: 1.5Gi
+          terminationMessagePath: /dev/termination-log
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 30
+    test: false
+    triggers:
+    - type: ConfigChange
+  status:
+    details:
+      causes:
+      - type: ConfigChange
+- kind: Service
+  apiVersion: v1
+  metadata:
+    name: build-tool-detector
+    labels:
+      service: build-tool-detector
+  spec:
+    ports:
+      - name: "8099"
+        protocol: TCP
+        port: 80
+        targetPort: 8099
+    selector:
+      service: build-tool-detector
+    type: ClusterIP
+    sessionAffinity: null
+parameters:
+- name: IMAGE
+  value: quay.io/openshiftio/rhel-fabric8-services-build-tool-detector
```

Comment:
maybe it is better to set here default image to CentOS based [0] because it is public and will enable to apply this template on any OS like local for dev purposes?

[0] - quay.io/openshiftio/fabric8-services-build-tool-detector

and in https://github.com/openshiftio/saas-openshiftio/pull/1127/files you override it to rhel based image for preview / prod

Reply:
Reasonable, thanks @riuvshin . Done in https://github.com/fabric8-services/build-tool-detector/pull/7/commits/357f98f540564d1386992dff081a157be0e5e126